File: projects-python/motion-led/motion-led.py
import RPi.GPIO as GPIO
import time
import sys
from Pubnub import Pubnub

pubnub = Pubnub(publish_key='demo', subscribe_key='demo')
channel = 'motionsensor'
message = 'Motion detected'

def callback(message):
    print(message)


GPIO.setmode(GPIO.BCM)
PIR_PIN = 4
GPIO.setup(PIR_PIN, GPIO.IN)

LED_PIN = 17
GPIO.setup(LED_PIN, GPIO.OUT)

def MOTION(PIR_PIN):
    print 'Motion Detected!'
        GPIO.output(LED_PIN, True)
            pubnub.publish(channel, message, callback=callback, error=callback)
# GPIO.output(LED_PIN, False)

def STILL(PIR_PIN):
    print 'No Motion Detected!'
        GPIO.output(LED_PIN, False)
# GPIO.output(LED_PIN, False)

print 'PIR Module Test (CTRL+C to exit)'
time.sleep(2)
print 'Ready'

try:
    GPIO.add_event_detect(PIR_PIN, GPIO.RISING, callback=MOTION)
        while 1:
time.sleep(100)

except KeyboardInterrupt:
    print ' Quit'



try:
    GPIO.add_event_detect(PIR_PIN, GPIO.FALLING, callback=STILL)
        while 1:
time.sleep(100)

except KeyboardInterrupt:
    print ' Quit'
GPIO.cleanup()

File: projects-nodejs/motion-sensor/motion-led.js
/* 
 * Using a PIR motion sensor with Pi (add-on to motion.js)
 * When motion is detected a LED lights up as a visual signal
 */

var raspi = require('raspi-io');
var five = require('johnny-five');
var board = new five.Board({io: new raspi()});

// Initializing PubNub
var channel = 'motionsensor';
var pubnub = require('pubnub').init({
	publish_key: 'demo',
	subscribe_key: 'demo'
});

board.on('ready', function() {
	console.log('board is ready');

	// Create hardware instances
	var led = new five.IR.LED('P1-11'); //pin 11 (GPIO 17)
	var motion = new five.IR.Motion('P1-7'); //pin 7 (GPIO 4)

	// 'calibrated' occurs once, at the beginning of a session,
	motion.on('calibrated', function() {
		console.log('calibrated');
	});

	// 'motionstart' events are fired when the 'calibrated'
	// proximal area is disrupted, generally by some form of movement
	motion.on('motionstart', function() {
		console.log('motionstart');
		led.on();

		pubnub.publish({
			channel: channel,
			message: 'Motion detected',
			callback: function(m) {console.log(m);}
		});
	});

	// 'motionend' events are fired following a 'motionstart' event
	// when no movement has occurred in X ms
	motion.on('motionend', function() {
		console.log('motionend');
		led.off();
	});
});